util/test/box_testutil.go

```diff
@@ -0,0 +1,146 @@
+package test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/transactions/logic"
+	"github.com/algorand/go-algorand/ledger/ledgercore"
+
+	indxLedger "github.com/algorand/indexer/processor/eval"
+	"github.com/stretchr/testify/require"
+)
+
+func getNameAndAccountPointer(t *testing.T, value *string, fullKey string, accts map[basics.Address]*ledgercore.AccountData) (basics.Address, string, *ledgercore.AccountData) {
+	require.NotNil(t, value, "cannot handle a nil value for box stats modification")
+	appIdx, name, err := logic.SplitBoxKey(fullKey)
+	account := appIdx.Address()
+	require.NoError(t, err)
+	acctData, ok := accts[account]
+	if !ok {
+		acctData = &ledgercore.AccountData{
+			AccountBaseData: ledgercore.AccountBaseData{},
+		}
+		accts[account] = acctData
+	}
+	return account, name, acctData
+}
+
+func addBoxInfoToStats(t *testing.T, fullKey string, value *string,
+	accts map[basics.Address]*ledgercore.AccountData, boxTotals map[basics.Address]basics.AccountData) {
+	addr, name, acctData := getNameAndAccountPointer(t, value, fullKey, accts)
+
+	acctData.TotalBoxes++
+	acctData.TotalBoxBytes += uint64(len(name) + len(*value))
+
+	boxTotals[addr] = basics.AccountData{
+		TotalBoxes:    acctData.TotalBoxes,
+		TotalBoxBytes: acctData.TotalBoxBytes,
+	}
+}
+
+func subtractBoxInfoToStats(t *testing.T, fullKey string, value *string,
+	accts map[basics.Address]*ledgercore.AccountData, boxTotals map[basics.Address]basics.AccountData) {
+	addr, name, acctData := getNameAndAccountPointer(t, value, fullKey, accts)
+
+	prevBoxBytes := uint64(len(name) + len(*value))
+	require.GreaterOrEqual(t, acctData.TotalBoxes, uint64(0))
+	require.GreaterOrEqual(t, acctData.TotalBoxBytes, prevBoxBytes)
+
+	acctData.TotalBoxes--
+	acctData.TotalBoxBytes -= prevBoxBytes
+
+	boxTotals[addr] = basics.AccountData{
+		TotalBoxes:    acctData.TotalBoxes,
+		TotalBoxBytes: acctData.TotalBoxBytes,
+	}
+}
+
+// BuildAccountDeltasFromKvsAndMods simulates keeping track of the evolution of the box statistics
+func BuildAccountDeltasFromKvsAndMods(t *testing.T, kvOriginals, kvMods map[string]*string) (
```

Comment:
I slightly prefer putting this in `writer_test.go`, or a `utils_test.go` file in the `writer` package. It's possible for files that don't end in `_test.go` to slip into a production binary, so unless they're really used across packages it is best to avoid using real packages.

[This is a long time pet peeve for me.](https://github.com/golang/go/issues/44086)

Reply:
I appreciate your valiant attempt to fight the machine. I'll investigate refactoring along the lines suggested. The most frustration for me during this PR has been working around go's import restrictions for testing.

Reply:
BTW. There is a significant refactor of testing code in the upcoming followup #1117. It might be worthwhile to punt the discussion of where tests should live exactly till that PR.

Reply:
In particular, as part of the refactor this file has been deleted:
<img width="859" alt="image" src="https://user-images.githubusercontent.com/291133/183566761-37b9f091-bc5b-4744-8513-c2e4b91ed772.png">

Reply:
@winder I've analyzed my refactoring for upcoming #1117. That one is not yet ready for review. However, one of the refactorings was doing exactly what you suggested. Moving `BuildAccountDeltasFromKvsAndMods` to [writer_test.go](https://github.com/algorand/indexer/blob/34887da8ca7bc9ed6d1d38166846c3003653c911/idb/postgres/internal/writer/writer_test.go#L1615). Thus I ask that we punt the final refactoring discussion to that PR.